WikiWatershed/model-my-watershed
# Set TR-55 to use NLCD11 2011 by default

## Overview

As noted in #3527, the NLCD19 + Soil combination for `RasterGroupedCount` would result in unstable values, reporting NODATA for cases that had real values.

While investigating this failure, it was observed that the NLCD11 2011 + Soil combination worked correctly, as before. After conferring with clients, it was decided that TR-55 should be switched back to NLCD11 2011, until further notice.

This code change allows the specification of a land layer override, but in the absence of an override will use the NLCD11 2011 layer. Since the UI does not specify any overrides, the NLCD11 2011 layer will be used for all cases.

Connects #3527 

### Demo

<img width="1512" alt="image" src="https://user-images.githubusercontent.com/1430060/164339691-8fa08376-c8d2-47c4-a789-d716f8b52a50.png">

### Notes

This bug may have implications for GWLF-E as well. Will continue investigating the original issue, but this should alleviate the issue for TR-55 users.

## Testing Instructions

* On `master`, ensure you are logged in so that your work is saved
* Make a TR-55 project with this shape: [rivera.geojson.txt](https://github.com/WikiWatershed/model-my-watershed/files/8522711/rivera.geojson.txt)
* Draw some modifications on it
  - [ ] Ensure it fails
* Check out this branch
* Run migrations
  ```shell
  ./scripts/manage migrate
  ```
* Restart Celery
  ```shell
  vagrant ssh worker -c 'sudo service celeryd restart'
  ```
* Go back to your saved project
